src/main.cpp

```diff
@@ -2837,6 +2844,9 @@ bool static ConnectTip(CValidationState& state, const CChainParams& chainparams,
     for(unsigned int i=0; i < pblock->vtx.size(); i++)
         txChanged.push_back(std::make_tuple(pblock->vtx[i], pindexNew, i));
 
+    // update mempool stats
+    CStats::DefaultStats()->addMempoolSample(mempool.size(), mempool.DynamicMemoryUsage(), mempool.GetMinFee(GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000).GetFeePerK());
```

Comment:
Can you please do something with this long line? It is there ~3 times and does a simple thing - worth a better method to add a sample?